bindings/java/src/org/sleuthkit/datamodel/HostManager.java

```diff
@@ -133,15 +133,116 @@ Host createHost(String name, CaseDbTransaction trans) throws TskCoreException {
 					throw new TskCoreException(String.format("Error adding host with name = %s and unable to rollback", name), ex);
 				}
 			}
-			
+
 			// It may be the case that the host already exists, so try to get it.
 			Optional<Host> optHost = getHost(name, connection);
 			if (optHost.isPresent()) {
 				return optHost.get();
 			}
 			throw new TskCoreException(String.format("Error adding host with name = %s", name), ex);
 		}
-	}	
+	}
+
+	/**
+	 * Updates host in database based on the host object provided.
+	 *
+	 * @param newHost The host to be updated.
+	 *
+	 * @return The newly returned host.
+	 *
+	 * @throws TskCoreException
+	 * @throws IllegalArgumentException
+	 */
+	public Host updateHost(Host newHost) throws TskCoreException, IllegalArgumentException {
+		if (newHost == null) {
+			throw new IllegalArgumentException("No host argument provided.");
+		} else if (newHost.getName() == null) {
+			throw new IllegalArgumentException(String.format("Host with id %d has no name", newHost.getId()));
+		}
+
+		CaseDbConnection connection = this.db.getConnection();
```

Comment:
Need to close this or move to the try()

Reply:
Ok.  Should be in a try-with-resources now.